### PR TITLE
Allow a custom Logger to be used as the backend for boa_runtime::Console

### PR DESCRIPTION
**Pull request by hansl**
There are some cleanup that could be done in the number of lines of codes, but this was the smallest change I could do to make the change.

This comes from a need to implement `Console` with using `tracing` as the backend. Please note that there is an issue here where multiple logs in the same function (e.g. `console.trace()`) would output into multiple log entries. This should probably be fixed in a following PR (as this one is already large enough).

### CODE REVIEW

**Review by jedel1043 — APPROVED** [review]
Looks perfect to me!

File: core/runtime/src/lib.rs [hunk 1]
```diff
@@ -56,6 +56,9 @@ mod console;
 #[doc(inline)]
 pub use console::Console;
 
+#[doc(inline)]
+pub use console::Logger;
```
Comment (jasonwilliams) [inline core/runtime/src/lib.rs:60]: Can you not do `pub use console::{Console, Logger}` here?

**Review by jasonwilliams — APPROVED** [review]
LGTM (apart from comment), thanks @hansl

**Comment by hansl** [comment]
I made the change, PTAL. Sorry I asked for a re-review from @jedel1043 by mistake.

**Review by jedel1043 — APPROVED** [review]

**Pull request merged**